libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -140,6 +140,11 @@ AC_WPNav::AC_WPNav(const AP_InertialNav& inav, const AP_AHRS_View& ahrs, AC_PosC
     _flags.recalc_wp_leash = false;
     _flags.new_wp_destination = false;
     _flags.segment_type = SEGMENT_STRAIGHT;
+
+    // sanity check loiter speed
+    if (_loiter_speed_cms <= 0) {
+        _loiter_speed_cms = WPNAV_LOITER_SPEED;
```

Comment:
It's a matter of choice, but I would check for less than WPNAV_LOITER_SPEED_MIN since we use it as limit in other places. Also, more importantly, if the check is true, I would set it to WPNAV_LOITER_SPEED_MIN, otherwise the user might be surprised by a much bigger speed.